Added Key throttling support

munezbn opened the pull request:
We can configure key throttling now which will be useful for low-end platform.
Also added Threadsafe QUEUE sdk module

munezbn committed:
Added Key throttling support

We can configure key throttling now which will be useful for low-end platform.
Also added Threadsafe QUEUE sdk module

shefayaparwin commented on ReactSkia/core_modules/RSkSpatialNavigator.cpp:
```diff
@@ -61,7 +77,10 @@ void RSkSpatialNavigator::updateSpatialNavigatorState(NavigatorStateOperation op
-                sendNotificationWithEventType("focus", candidateData.tag);
+                sendNotificationWithEventType("focus", candidateData.tag, completeCallback_);
+#if ENABLE(FEATURE_KEY_THROTTLING)
+                RSkInputEventManager::getInputKeyEventManager()->onEventEmit();
```
will it be good to trigger onEventEmit first before sending EVent

shefayaparwin commented on ReactSkia/core_modules/RSkSpatialNavigator.cpp:
```diff
@@ -411,7 +434,10 @@ void RSkSpatialNavigator::updateFocusCandidate(RSkComponent* focusCandidate){
     sendNotificationWithEventType("blur", currentFocus_->getComponentData().tag);
```
throttling condition not needed Here ?

shefayaparwin commented on ReactSkia/core_modules/RSkInputEventManager.cpp:
```diff
@@ -26,40 +30,104 @@ RSkInputEventManager::RSkInputEventManager(){
 }
 
+void RSkInputEventManager::onEventComplete() {
+  (activeInputClients_ > 0) ? ((--activeInputClients_ == 0) ? sem_post(&keyEventPost_) : false) : false;
```
Will there be a possibility : activeInputClients_ will be > 1 and possibly blocks the key processing

munezbn replied:
The idea of key throttling is to block the key processing until previous key is completely processed by all the active clients.

munezbn committed:
Addressed Review comments

Moved onEventEmit() call inside sendNotificationWithEventType() for reusability.
